crates/comms/src/livekit_web.rs

```diff
@@ -0,0 +1,279 @@
+    debug!(">> lk connect async : {}", remote_address);
+
+    let url = Uri::try_from(remote_address).unwrap();
+    let address = format!(
+        "{}://{}{}",
+        url.scheme_str().unwrap_or_default(),
+        url.host().unwrap_or_default(),
+        url.path()
+    );
+    let params = HashMap::from_iter(url.query().unwrap_or_default().split('&').flat_map(|par| {
+        par.split_once('=')
+            .map(|(a, b)| (a.to_owned(), b.to_owned()))
+    }));
+    debug!("{:?}", params);
+    let token = params.get("access_token").cloned().unwrap_or_default();
+
+    // In WASM, we can't block or create threads, so we just spawn the async task
+    spawn_local(async move {
+        if let Err(e) = run_livekit_session(transport_id, &address, &token, app_rx, sender).await {
+            error!("LiveKit session error: {:?}", e);
+        }
+    });
+
+    Ok(())
+}
+
+async fn run_livekit_session(
+    transport_id: Entity,
+    address: &str,
+    token: &str,
+    app_rx: Arc<Mutex<Receiver<NetworkMessage>>>,
+    sender: Sender<PlayerUpdate>,
+) -> Result<(), anyhow::Error> {
+    let room = connect_room(address, token)
+        .await
+        .map_err(|e| anyhow::anyhow!("Failed to connect room: {:?}", e))?;
+
+    let sender_clone = sender.clone();
+
+    // Set up event handler
+    let event_handler = Closure::wrap(Box::new(move |event: JsValue| {
+        let sender = sender_clone.clone();
+
+        spawn_local(async move {
+            handle_room_event(event, transport_id, sender).await;
+        });
+    }) as Box<dyn FnMut(JsValue)>);
+
+    set_room_event_handler(&room, &event_handler)
+        .map_err(|e| anyhow::anyhow!("Failed to set event handler: {:?}", e))?;
+
+    // Keep the closure alive
+    event_handler.forget();
+
+    // Microphone is handled entirely in JavaScript
+
+    // Handle outgoing messages
+    let mut app_rx = app_rx.lock().await;
+    loop {
+        let message = app_rx.recv().await;
+        let Some(outgoing) = message else {
+            debug!("App pipe broken, exiting loop");
+            break;
+        };
+
+        let destinations = if let Some(address) = outgoing.recipient {
+            js_sys::Array::of1(&JsValue::from_str(&format!("{:#x}", address)))
+        } else {
+            js_sys::Array::new()
+        };
+
+        if let Err(e) = publish_data(
+            &room,
+            &outgoing.data,
+            !outgoing.unreliable,
+            destinations.into(),
+        )
+        .await
+        {
+            warn!("Failed to publish data: {:?}", e);
+            break;
+        }
+    }
+
+    close_room(&room)
+        .await
+        .map_err(|e| anyhow::anyhow!("Failed to close room: {:?}", e))?;
+
+    Ok(())
+}
+
+async fn handle_room_event(event: JsValue, transport_id: Entity, sender: Sender<PlayerUpdate>) {
+    // Parse the event from JavaScript
+    // This is a simplified version - you'll need to properly parse the JavaScript event object
```

Comment:
all this should be handled by serde_wasm_bindgen - you can make a rust struct with `#[derive(serde::Deserialize)]` and `#[serde(rename_all='camelCase')]` and then deserialize a JsValue into it with `serde_wasm_bindgen::from_value(js_value)`, we do this in dcl_wasm already.